### PR TITLE
[Add] props description, descriptions to the component.

### PR DESCRIPTION
**Pull request by khushal87**
**What kind of change does this PR introduce?**
This PR introduces adding descriptions and props descriptions to the components for auto-generation of docs in the future.

**Did you add tests for your changes?**
Yes, I have added and configured the tests accordingly.

**If relevant, did you update the documentation?**
Not required.

**Summary**
This PR added descriptions and props description to the components except for the class-based components:
- Image
- Input
- Searchbar
- Tooltip

**Does this PR introduce a breaking change?**
No.

**Other information**
Nothing as such. I have used a common pattern for commenting on the descriptions in the components.

**Tags**
#gsoc #docs-autogen #khushal87

### CODE REVIEW

**Comment by arpitBhalla** [comment]
@khushal87 Could you add this feature too? https://github.com/react-native-elements/react-native-elements/pull/3105#discussion_r657893869

**Comment by khushal87** [comment]
@arpitBhalla This is not decided yet, I will have to talk to mentors. Adding all this will unnecessarily make the component long with descriptions in my opinion.

**Comment by arpitBhalla** [comment]
@khushal87 Its already decided in #3004

**Comment by khushal87** [comment]
I can't see any mentors/admin feedback on it.

**Comment by arpitBhalla** [comment]
definitely adding single line for preview would not make description lengthy

File: src/AirbnbRating/AirbnbRating.tsx [hunk 1]
```diff
@@ -7,6 +7,11 @@ import { RneFunctionComponent } from '../helpers';
 
 export type TapRatingProps = RatingProps;
 
+/**Ratings are used to collect measurable feedback from users.
```
Comment (dhruvdutt) [inline src/AirbnbRating/AirbnbRating.tsx:10]:
```suggestion
/** Ratings are used to collect measurable feedback from users.
```

File: src/Avatar/Accessory.tsx [hunk 2]
```diff
@@ -14,10 +14,15 @@ import { RneFunctionComponent } from '../helpers';
 
 export type AccessoryProps = Partial<IconProps> &
   Partial<ImageProps> & {
+    /**Add underlay color to the accessory of avatar. */
```
Comment (dhruvdutt) [inline src/Avatar/Accessory.tsx:17]:
```suggestion
    /** Add underlay color to the accessory of avatar. */
```

Comment (dhruvdutt) [reply src/Avatar/Accessory.tsx:17]: @khushal87 Can you make this change everywhere? You can find and replace all.

Comment (khushal87) [reply src/Avatar/Accessory.tsx]: @dhruvdutt Done, thanks for pointing it out. 🙌

**Review by flyingcircle — APPROVED** [review]

**Comment by arpitBhalla** [comment]
@dhruvdutt Can we add preview images and documentation links to description, it would be easy for developer to quickly access docs.

**Comment by khushal87** [comment]
@arpitBhalla The problem with this is that it will also be visible in documentation after parse unless and until we remove it by adding additional conditions. I don't think making the code complex is a good thing, where we already have an underlying props filter complexity which we would be having. Docs are enough to show what the components are used for. This is my opinion. @dhruvdutt what do you say?

**Comment by arpitBhalla** [comment]
> I don't think making the code complex

removing 2 lines would not be complex, you can do it after generating markdown files, I guess 🙄

**Review by dhruvdutt — APPROVED** [review]

**Comment by dhruvdutt** [comment]
@arpitBhalla Adding preview image & description link can be picked up separately as an enhancement later. Let's track it at #3115 and add it for all components together since this would need some changes in the comments parsing for filtering during auto-generation of docs.

**Pull request merged**